orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -634,6 +634,56 @@ at::Tensor slice_Tensor(
     self.options());
 }
 
+// aten::argmax.out(Tensor self, int? dim=None, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
+at::Tensor& argmax_out(
+const at::Tensor& self,
+c10::optional<int64_t> dim,
+bool keepdim,
+// *,
+at::Tensor& out) {
+  ORT_LOG_FN(self, dim, keepdim, out);
+
+  if (
+    !IsSupportedType(self, {at::kLong,at::kShort,at::kHalf,at::kBFloat16,at::kFloat,at::kByte,at::kInt,at::kDouble})) {
+    return at::native::call_fallback_fn<
+    &at::native::cpu_fallback,
+    ATEN_OP(argmax_out)>::call(self, dim, keepdim, out);
+  }
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto ort_input_self =
+    create_ort_value(invoker, dim.has_value() ? self : self.reshape({-1}));
+
+  // Remove this hand signature once the generator can support this one line below.
+  int64_t l_axis = dim.has_value() ? *dim : 0;
+
+  NodeAttributes attrs(2);
+  attrs["axis"] = create_ort_attribute(
+  "axis", l_axis, at::ScalarType::Int);
+  attrs["keepdims"] = create_ort_attribute(
+  "keepdims", keepdim, at::ScalarType::Int);
+
+  std::vector<OrtValue> ort_outputs_0_ArgMax(1);
+
+  auto status = invoker.Invoke("ArgMax", {
+  std::move(ort_input_self),
+  }, ort_outputs_0_ArgMax, &attrs);
+
+  if (!status.IsOK())
+  throw std::runtime_error(
+  "ORT return failure status:" + status.ErrorMessage());
+
+  at::TensorOptions tensor_options = out.options().dtype(at::kLong);
```

Comment:
Do you need `dtype(at::kLong)`? I thought out was already of type `at::kLong`?

Reply:
you are correct, that is not needed. Thank you!